.github/workflows/build.yml

```diff
@@ -1,6 +1,10 @@
 name: 'Build project'
 
-on: [push]
+on:
+  push:
+    branches: [master]
```

Comment:
this means PR that target master?

Reply:
This part means it should run on any `push` to master, does not matter the reason/origin. So main use case here after a merge of a pull request.
```yaml
on:
  push:
    branches: [master]
```

And this means this workflow should run for pull requests which target `master`. So one opening a PR against `master`.
```yaml
on:
  pull_request:
    branches: [master]
```

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#on